### ISSUE TITLE
Very slow animations (especially compared to KDE and GNOME)

### ISSUE THREAD

**Issue by Freemason-12** [linuxmint/cinnamon#11373]
```
 * Cinnamon version (5.6.5)
   - No daily build PPA
 * Distribution - Mint 21
 * Intel HD Graphics 520
 * 64 bit
 ```

### Issue
Animations of overview effect and Expo are slow as hell (very low FPS)

### Steps to reproduce
trigger overview effect or Expo

### Expected behaviour
Everything smooth

### Other information

https://user-images.githubusercontent.com/97738675/210048433-d56bb814-052e-4bdb-b2be-a29addbf00ad.mp4

### Configuration info, if applicable
Animation issue has been a very big problem for for a lot of time. Issue seems to persist since Cinnamon 4.6.7 but before Cinnamon had an option in settings > general for animation type, and setting this specific setting to 'something else that default solved the problem (at least partially because animations were smooth but some flickering happened sometimes). Now in Cinnamon 5.4 and later this option disappeared and animations got worse

**Comment by Freemason-12** [comment]
**Another video on Mint 20.3 with Cinnamon 5.2.7**
the setting I was talking about earlier above in _VSync method_. When set to _presentation time_ (default) it was like this:

https://user-images.githubusercontent.com/97738675/210053276-80d4acbe-4ce9-45d1-adee-ec2fd4cd856d.mp4

And it became smoother when setting it to a different parameter. But not in all cases:

https://user-images.githubusercontent.com/97738675/210053653-01507ee6-1e70-4e90-879d-fa4d61de07e4.mp4

**Comment by xhpohanka** [comment]
I have similar experience:

```
System:
  Kernel: 6.1.9-arch1-1 arch: x86_64 bits: 64 compiler: gcc v: 12.2.1 Desktop: Cinnamon v: 5.6.7
    tk: GTK v: 3.24.36 dm: LightDM Distro: Arch Linux
Machine:
  Type: Laptop System: LENOVO product: 20RH001PMC v: ThinkPad P43s serial: <superuser required>
    Chassis: type: 10 serial: <superuser required>
  Mobo: LENOVO model: 20RH001PMC v: SDK0J40697 WIN serial: <superuser required> UEFI: LENOVO
    v: N2IETA0W (1.78 ) date: 10/14/2022
Battery:
  ID-1: BAT0 charge: 51.5 Wh (100.0%) condition: 51.5/50.5 Wh (101.9%) volts: 12.9 min: 11.6
    model: LGC 02DL007 serial: <filter> status: full
CPU:
  Info: quad core model: Intel Core i7-8565U bits: 64 type: MT MCP arch: Comet/Whiskey Lake
    note: check rev: C cache: L1: 256 KiB L2: 1024 KiB L3: 8 MiB
  Speed (MHz): avg: 706 high: 744 min/max: 400/4600 cores: 1: 702 2: 701 3: 701 4: 701 5: 744
    6: 701 7: 700 8: 703 bogomips: 32012
  Flags: avx avx2 ht lm nx pae sse sse2 sse3 sse4_1 sse4_2 ssse3 vmx
Graphics:
  Device-1: Intel WhiskeyLake-U GT2 [UHD Graphics 620] vendor: Lenovo driver: i915 v: kernel
    arch: Gen-9.5 ports: active: DP-3,DP-6 off: eDP-1 empty: DP-1, DP-2, DP-5, HDMI-A-1, HDMI-A-2
    bus-ID: 00:02.0 chip-ID: 8086:3ea0
  Device-2: NVIDIA GP108GLM [Quadro P520] vendor: Lenovo driver: N/A arch: Pascal pcie:
    speed: Unknown lanes: 63 bus-ID: 3c:00.0 chip-ID: 10de:1d34
  Device-3: Chicony Integrated Camera (1280x720@30) type: USB driver: uvcvideo bus-ID: 1-8:6
    chip-ID: 04f2:b604
  Display: x11 server: X.org v: 1.21.1.6 driver: X: loaded: intel dri: i965 gpu: i915
    display-ID: :0 screens: 1
  Screen-1: 0 s-res: 3840x1200
  Monitor-1: not-matched mapped: DP2-1 pos: primary,left res: 1920x1200 dpi: 94
    diag: 611mm (24.04")
  Monitor-2: not-matched mapped: DP2-3 pos: right res: 1920x1200 dpi: 94 diag: 611mm (24.04")
  Monitor-3: eDP-1 mapped: eDP1 note: disabled model: BOE Display 0x07c9 res: 1920x1200 dpi: 163
    diag: 345mm (13.6")
  API: OpenGL Message: Unable to show GL data. Required tool glxinfo missing.
Audio:
  Device-1: Intel Cannon Point-LP High Definition Audio vendor: Lenovo driver: snd_hda_intel
    bus-ID: 1-2.3.4:40 v: kernel bus-ID: 00:1f.3 chip-ID: 0bda:402e chip-ID: 8086:9dc8
  Device-2: Realtek USB Audio type: USB driver: snd-usb-audio
  Sound API: ALSA v: k6.1.9-arch1-1 running: yes
  Sound Server-1: PulseAudio v: 16.1 running: no
  Sound Server-2: PipeWire v: 0.3.65 running: yes
Network:
  Device-1: Intel Cannon Point-LP CNVi [Wireless-AC] driver: iwlwifi v: kernel bus-ID: 00:14.3
    chip-ID: 8086:9df0
  IF: wlp0s20f3 state: down mac: <filter>
  Device-2: Intel Ethernet I219-V vendor: Lenovo driver: e1000e v: kernel port: N/A
    bus-ID: 00:1f.6 chip-ID: 8086:15be
  IF: enp0s31f6 state: up speed: 1000 Mbps duplex: full mac: <filter>
  Device-3: Realtek RTL8153 Gigabit Ethernet Adapter type: USB driver: r8152 bus-ID: 2-2.4:12
    chip-ID: 0bda:8153
  IF: enp0s20f0u2u4 state: up speed: 1000 Mbps duplex: full mac: <filter>
  IF-ID-1: tun0 state: unknown speed: 10 Mbps duplex: full mac: N/A
Bluetooth:
  Device-1: Intel Bluetooth 9460/9560 Jefferson Peak (JfP) type: USB driver: btusb v: 0.8
    bus-ID: 1-10:12 chip-ID: 8087:0aaa
  Report: bt-adapter ID: hci0 rfk-id: 6 state: up address: <filter>
Drives:
  Local Storage: total: 1.82 TiB used: 562.12 GiB (30.2%)
  ID-1: /dev/nvme0n1 vendor: Kingston model: SNV2S2000G size: 1.82 TiB speed: 63.2 Gb/s lanes: 4
    serial: <filter> temp: 47.9 C
Partition:
  ID-1: / size: 105.18 GiB used: 51.13 GiB (48.6%) fs: ext4 dev: /dev/nvme0n1p3
  ID-2: /boot size: 246.1 MiB used: 152.1 MiB (61.8%) fs: vfat dev: /dev/nvme0n1p2
Swap:
  ID-1: swap-1 type: partition size: 23.44 GiB used: 0 KiB (0.0%) priority: -2 dev: /dev/nvme0n1p6
Sensors:
  System Temperatures: cpu: 53.0 C pch: 52.0 C mobo: N/A
  Fan Speeds (RPM): fan-1: 3584
Repos:
  Packages: pm: pacman pkgs: 1461
  Active pacman repo servers in: /etc/pacman.conf
    1: http: //repo.archlinuxcn.org/$arch
    2: https: //download.sublimetext.com/arch/dev/x86_64
    3: https: //repo.herecura.eu/herecura/x86_64
  Active pacman repo servers in: /etc/pacman.d/mirrorlist
    1: http: //gluttony.sin.cvut.cz/arch/$repo/os/$arch
    2: http: //ftp.linux.cz/pub/linux/arch/$repo/os/$arch
    3: http: //mirror.dkm.cz/archlinux/$repo/os/$arch
    4: http: //mirror.vpsfree.cz/archlinux/$repo/os/$arch
    5: http: //ftp.fi.muni.cz/pub/linux/arch/$repo/os/$arch
    6: http: //mirrors.nic.cz/archlinux/$repo/os/$arch
Info:
  Processes: 337 Uptime: 17h 38m Memory: 23.14 GiB used: 6.56 GiB (28.3%) Init: systemd v: 252
  default: graphical Compilers: gcc: 12.2.1 alt: 11 clang: 15.0.7 Client: Cinnamon v: 5.6.7
  inxi: 3.3.24

**Comment by Freemason-12** [comment]
cinnamon 5.6.8 is out already. Any progress on this? (this and only this makes linux mint unusable for me)

**Comment by AbdullahPrime** [comment]
I'm using KDE Plasma on-top of mint 21 and KDE works buttery smooth but Cinnamon's glitchy animations are a REAL PAIN to use ....

**Comment by Freemason-12** [comment]
Please reopen the issue, the merge does not address it.

**Comment by dacampsss** [comment]
I tried Mint right now (live USB) and while Cinnamon looks really great (and I think it'd be better to use Cinnamon with Mint than with other distros), animations really are not looking smooth here. Those are my specs:

```
System:
  Host: debian Kernel: 6.12.27-tkg-eevdf arch: x86_64 bits: 64
  Desktop: KDE Plasma v: 6.3.4 Distro: Debian GNU/Linux trixie/sid
Machine:
  Type: Desktop System: DELTA- product: N/A v: N/A
    serial: <superuser required>
  Mobo: DELTA- model: B85M2K v: V1.0 serial: <superuser required>
    UEFI: American Megatrends v: 5.6.5 date: 04/10/2024
CPU:
  Info: quad core Intel Xeon E3-1241 v3 [MT MCP] speed (MHz): avg: 2395
    min/max: 800/3900
Graphics:
  Device-1: Advanced Micro Devices [AMD/ATI] Ellesmere [Radeon RX
    470/480/570/570X/580/580X/590] driver: amdgpu v: kernel
  Device-2: Microdia HoverCam Solo Spark Audio driver: uvcvideo type: USB
  Display: wayland server: X.org v: 1.21.1.16 with: Xwayland v: 24.1.6
    compositor: kwin_wayland driver: X: loaded: amdgpu
    unloaded: fbdev,modesetting,vesa dri: radeonsi gpu: amdgpu
    resolution: 1920x1080~75Hz
  API: OpenGL v: 4.6 compat-v: 4.5 vendor: amd mesa v: 25.0.5-1 renderer: AMD
    Radeon RX 570 Series (radeonsi polaris10 ACO DRM 3.61 6.12.27-tkg-eevdf)
  Info: Tools: api: clinfo, eglinfo, glxinfo, vulkaninfo
    de: kscreen-console,kscreen-doctor gpu: lact,radeontop wl: wayland-info
    x11: xdriinfo, xdpyinfo, xprop, xrandr
Network:
  Device-1: Realtek RTL8111/8168/8211/8411 PCI Express Gigabit Ethernet
    driver: r8169
Drives:
  Local Storage: total: 1.12 TiB used: 772.23 GiB (67.1%)
Info:
  Memory: total: 16 GiB available: 15.56 GiB used: 7.9 GiB (50.8%)
  Processes: 350 Uptime: 28m Shell: Bash inxi: 3.3.38

```

**Comment by maurocsound** [comment]
I just tried Linux Mint 22.1 Cinnamon edition (first with a live USB, then by installing the os on the ssd) and I also noticed that, at least on my notebook with i7-4720HQ CPU and Intel HD Graphics 4600, using a 1920x1080 resolution, there are annoying lags during windows movements and animations. Yeah, the GPU is not that great, but the same notebook in Windows 10 behaves much better (of course, same Full-HD resolution). In Gnome (tried using the latest Ubuntu 24.04) it behaves very smooth. With XFCE it behaves much smoother too. So, I think there could be something to optimize in the rendering code of Cinnamon desktop manager. Maybe it's a vsync issue? I didn't find an option to enable/disable vsync, so I don't know.
